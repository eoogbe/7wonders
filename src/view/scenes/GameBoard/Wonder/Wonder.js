/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { compose } from 'ramda';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from 'model/actions';
import { getCanBuildFree } from 'model/selectors/players';
import { getShields } from 'presentation/military';
import { getMoney } from 'presentation/players';
import { getWonderStages } from 'presentation/wonder';
import './Wonder.css';
import WonderStage from './WonderStage';

const renderStage = (stageActions) => (stage, idx) => (
  <WonderStage
    key={idx}
    {...stageActions}
    {...stage}
    index={idx}
  />
);

const Wonder = () => {
  const money = useSelector(getMoney);
  const shields = useSelector(getShields);
  const canBuildFree = useSelector(getCanBuildFree);
  const stages = useSelector(getWonderStages);

  const dispatch = useDispatch();
  const showWonderCallout = compose(dispatch, actions.showWonderCallout);

  return (
    <div>
      <div className="GameBoardWonder-tokenContainer">
        <div className="GameBoardWonder-token">{money}</div>
        <div className="GameBoardWonder-token">{shields}</div>
        {canBuildFree &&
          <div className="GameBoardWonder-token">Can build free structure</div>}
      </div>
      <ol className="GameBoardWonder-stageList">
        {stages.map(renderStage({ showWonderCallout }))}
      </ol>
    </div>
  );
};

export default Wonder;
