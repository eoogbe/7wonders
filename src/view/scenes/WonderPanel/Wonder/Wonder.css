/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.WonderPanelWonder {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
}

.WonderPanelWonder-currentPlayer {
  font-size: 1.25em;
  font-weight: bold;
}

.WonderPanelWonder-name {
  font-size: 1rem;
  display: flex;
  align-items: center;
  width: 10em;
  height: 6em;
  padding: 0.25em 0.5em;
  border-style: solid;
  border-color: #000;
  border-width: 0 1px 0 1px;
  cursor: pointer;
}
