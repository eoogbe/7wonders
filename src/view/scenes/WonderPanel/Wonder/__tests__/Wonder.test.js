/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Wonder from '../Wonder';

const NAME = 'Pyramids of Giza (A)';

const PLAYER_ID = 'PLAYER1';

it('renders without crashing', () => {
  const tree =
    renderer.create(
      <Wonder name={NAME} playerId={PLAYER_ID} isCurrentPlayer={false} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the current player', () => {
  const tree =
    renderer.create(
      <Wonder name={NAME} playerId={PLAYER_ID} isCurrentPlayer={true} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('view the wonder chooser', () => {
  const viewWonderChooser = jest.fn();
  const tree = shallow(
    <Wonder
      name={NAME}
      playerId={PLAYER_ID}
      isCurrentPlayer={false}
      viewWonderChooser={viewWonderChooser}
    />
  );
  tree.find('.WonderPanelWonder-name').simulate('click');
  expect(viewWonderChooser).toHaveBeenCalledWith(PLAYER_ID);
});
