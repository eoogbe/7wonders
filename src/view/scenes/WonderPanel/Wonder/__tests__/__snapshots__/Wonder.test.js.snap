// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the current player 1`] = `
<li
  className="WonderPanelWonder"
>
  <div
    className="WonderPanelWonder-currentPlayer"
  >
    player
  </div>
  <div
    className="WonderPanelWonder-name"
    onClick={[Function]}
  >
    Pyramids of Giza (A)
  </div>
</li>
`;

exports[`renders without crashing 1`] = `
<li
  className="WonderPanelWonder"
>
  <div
    className="WonderPanelWonder-name"
    onClick={[Function]}
  >
    Pyramids of Giza (A)
  </div>
</li>
`;
